### PR TITLE
Work on Set2. Up to Question 6

### PR DESCRIPTION
**Pull request by Thoth19**
Is this how this works?

### CODE REVIEW

File: config_parser_test.cc [hunk 1]
```diff
@@ -1,11 +1,43 @@
 #include "gtest/gtest.h"
 #include "config_parser.h"
-
+//this is an edit
```
Comment (algae-elbaum) [inline config_parser_test.cc]: What a silly comment :P

File: config_parser_test.cc [hunk 2]
```diff
@@ -1,11 +1,43 @@
...
+TEST_F(NginxStringConfigTest, SimpleBadTextConfig){
+    EXPECT_FALSE(ParseString("foo bar"));
+}
+TEST_F(NginxStringConfigTest, NestedConfig){
+    EXPECT_TRUE(ParseString(
+        "foo { "
+        "bar {"
+         "baz qux;"
+         "}"
+         "}"));
```
Comment (algae-elbaum) [inline config_parser_test.cc]: There isn't even indentation in this string. wow

File: config_parser_test.cc [hunk 3]
```diff
@@ -1,11 +1,43 @@
...
+}
```
Comment (algae-elbaum) [inline config_parser_test.cc:44]: All I can do is be pedantic about little style things. Other than that our code and the stuff from class is all just the same. Argh

File: config_parser_test.cc (hunk 1 above)
Comment (Thoth19) [inline config_parser_test.cc]: That's true. I'll get rid of that.

File: config_parser_test.cc (hunk 2 above)
Comment (Thoth19) [inline config_parser_test.cc]: I was writing fast. This isn't even an $\alpha$ release.

File: config_parser_test.cc (hunk 3 above)
Comment (Thoth19) [inline config_parser_test.cc:44]: That seems good then. I love how we are literally typing at each other from two feet away.

**Pull request merged**